src/terrain/routes/schemas/resource_usage_api.clj

```diff
@@ -0,0 +1,117 @@
+(def ListEventsSummary "List all events")
+(def ListEventsDescription "List all events in the system for all users")
+(def ListUserEventsSummary "List events for a user")
+(def ListUserEventsDescription "List all fo the events in the system for a single user")
+(def GetEventSummary "Get info about a single event")
+(def GetEventDescription "Returns information about a single event in the system")
+(def UpdateEventSummary "Update an event")
+(def UpdateEventDescription "Updates a single event in the system")
+(def DeleteEventSummary "Deletes an event")
+(def DeleteEventDescription "Deletes a single event in the system")
+
+(def WorkerID (describe UUID "The UUID assigned to a worker"))
+(def EventID (describe UUID "The UUID assigned to an event"))
+(def CPUHoursTotalID (describe UUID "The assigned to the record containing the CPU hours total"))
+(def HoursNumber (describe Long "The number of CPU hours"))
+(def Username (describe String "A username in the system"))
+
+(defschema Worker
+  {:id                      WorkerID
+   :name                    (describe String "The name of the worker")
+   :added_on                (describe String "When the worker was added to the system. Automatically set by the system")
+   :active                  (describe Boolean "Whether the worker is available for processing items. Set by the worker")
+   :activation_expires_on   (describe (maybe String) "When the worker's activation expires")
+   :deactivated_on          (describe (maybe String) "When the worker was deactivated. Automatically set by the system")
+   :activated_on            (describe (maybe String) "When the worker was activated. Automatically set by the system")
+   :getting_work            (describe Boolean "Whether the worker is claiming an event. Set by the worker")
+   :getting_work_expires_on (describe (maybe String) "When the worker must stop trying to claim an event. Automatically set by the system")
+   :getting_work_on         (describe (maybe String) "When the worker started claiming an event. Automatically set by the system")
+   :working                 (describe Boolean "Whether the worker is processing a event. Set by the worker")
+   :working_on              (describe (maybe String) "When the worker began processing the event")
+   :last_modified           (describe String "When the worker's record in the database was last modified")})
+
+(defschema UpdateWorker
+  (into {}
+        (for [keyval (select-keys
+                      Worker
+                      [:name
+                       :active
+                       :activation_expires_on
+                       :deactivated_on
+                       :getting_work
+                       :getting_work_on
+                       :getting_work_expires_on
+                       :working
+                       :working_on])]
+          [(optional-key (key keyval)) (val keyval)])))
```

Comment:
I think this could be handled by `schema-tools.core/optional-keys`.
For example:
https://github.com/cyverse-de/terrain/blob/09097dc798d0cbe519ba65e511c97dfb20a6d0c8/src/terrain/routes/schemas/filesystem/tickets.clj#L11-L14